src/sardana/pool/poolacquisition.py

```diff
@@ -716,7 +716,7 @@ def action_loop(self):
                 if is_value_error(value):
                     self.error("Loop final read value error for: %s" %
                                acquirable.name)
-                    acquirable.put_value(value)
+                    acquirable.extend_value_buffer(None, propagate=2)
```

Comment:
Your proposal here won't work because the `extend_value_buffer` code would try `len(None)` which would fail.
Could you clarify why you set the None only at the end of the action and not in the  main loop of the action?

Regardless of the above doubts, I would leave the code without changes in case of the hardware acquisition action. Intermediate exceptions in the read methods during the acquisition action does not precisely mean that the acquisition is failed. We base this decision on the channel's state. I can imagine a situation when one of the intermediate read method raises an exception, but then the subsequent executions correctly retrieve the data. The `put_value` call is there to indicate the error via the `Value` attribute.